src/objects/Line.js

```diff
@@ -35,6 +35,67 @@ Line.prototype = Object.assign( Object.create( Object3D.prototype ), {
 
 	isLine: true,
 
+	computeLineDistances: ( function () {
+
+		var start = new Vector3();
+		var end = new Vector3();
+
+		return function computeLineDistances() {
+
+			var distance = 0;
+			var geometry = this.geometry;
+
+			if ( geometry.isBufferGeometry ) {
+
+				// we assume non-indexed geometry
+
+				var positionAttribute = geometry.attributes.position;
+				var lineDistances = [];
+
+				for ( var i = 0, l = positionAttribute.count; i < l; i ++ ) {
+
+					if ( i > 0 ) {
+
+						start.fromBufferAttribute( positionAttribute, i - 1 );
+						end.fromBufferAttribute( positionAttribute, i );
+
+						distance += start.distanceTo( end );
+
+					}
+
+					lineDistances.push( distance );
```

Comment:
I wish we did not always create these JS arrays unnecessarily. It is a pattern we use over-and-over...